integration/networking/bridge_test.go

```diff
@@ -0,0 +1,334 @@
+		{
+			name: "IPv6 internal network",
+			bridgeOpts: []func(*types.NetworkCreate){
+				network.WithInternal(),
+			},
+		},
+	}
+
+	for tcID, tc := range testcases {
+		t.Run(tc.name, func(t *testing.T) {
+			ctx := testutil.StartSpan(ctx, t)
+
+			bridgeName := fmt.Sprintf("testnet-ip6ll-%d", tcID)
+			network.CreateNoError(ctx, t, c, bridgeName, append(tc.bridgeOpts,
+				network.WithDriver("bridge"),
+				network.WithIPv6(),
+				network.WithIPAM("fdf1:a844:380c:b247::/64", "fdf1:a844:380c:b247::1"),
+				network.WithOption("com.docker.network.bridge.name", bridgeName))...)
+			defer network.RemoveNoError(ctx, t, c, bridgeName)
+
+			ctr1Name := sanitizeCtrName(t.Name() + "-ctr1")
+			id1 := container.Run(ctx, t, c,
+				container.WithName(ctr1Name),
+				container.WithImage("busybox:latest"),
+				container.WithCmd("top"),
+				// Link-local address is derived from the MAC address, so we need to
+				// specify one here to hardcode the SLAAC LL address pinged below.
+				container.WithMacAddress("02:42:ac:11:00:02"),
+				container.WithNetworkMode(bridgeName))
+			defer c.ContainerRemove(ctx, id1, types.ContainerRemoveOptions{
+				Force: true,
+			})
+
+			ctr2Name := sanitizeCtrName(t.Name() + "-ctr2")
+			attachCtx, cancel := context.WithTimeout(ctx, 5*time.Second)
+			defer cancel()
+			res := container.RunAttach(attachCtx, t, c,
+				container.WithName(ctr2Name),
+				container.WithImage("busybox:latest"),
+				container.WithCmd("ping", "-c1", "-W3", "-6", "fe80::42:acff:fe11:2%eth0"),
+				container.WithNetworkMode(bridgeName))
+			defer c.ContainerRemove(ctx, res.ContainerID, types.ContainerRemoveOptions{
+				Force: true,
+			})
+
+			assert.Check(t, res.ExitCode == 0)
+			assert.Check(t, res.Stderr.Len() == 0)
+			assert.Check(t, is.Contains(res.Stdout.String(), "1 packets transmitted, 1 packets received"))
+		})
+	}
+}
+
+// TestBridgeINC makes sure two containers on two different bridge networks can't communicate with each other.
+func TestBridgeINC(t *testing.T) {
+	skip.If(t, testEnv.DaemonInfo.OSType == "windows")
+
+	ctx := setupTest(t)
+
+	d := daemon.New(t)
+	d.StartWithBusybox(ctx, t, "-D", "--experimental", "--ip6tables")
+	defer d.Stop(t)
+
+	c := d.NewClientT(t)
+	defer c.Close()
+
+	type bridgesOpts struct {
+		bridge1Opts []func(*types.NetworkCreate)
+		bridge2Opts []func(*types.NetworkCreate)
+	}
+
+	testcases := []struct {
+		name    string
+		bridges bridgesOpts
+		ipv6    bool
+		stdout  string
+		stderr  string
+	}{
+		{
+			name: "IPv4 non-internal network",
+			bridges: bridgesOpts{
+				bridge1Opts: []func(*types.NetworkCreate){},
+				bridge2Opts: []func(*types.NetworkCreate){},
+			},
+			stdout: "1 packets transmitted, 0 packets received",
+		},
+		{
+			name: "IPv4 internal network",
+			bridges: bridgesOpts{
+				bridge1Opts: []func(*types.NetworkCreate){network.WithInternal()},
+				bridge2Opts: []func(*types.NetworkCreate){network.WithInternal()},
+			},
+			stderr: "sendto: Network is unreachable",
```

Comment:
@corhere @thaJeztah Since #46603 has been merged, the error for such case has changed from "no response" to "network is unreachable".